src/util/DateYearDuration.cpp

```diff
@@ -150,14 +149,29 @@ DateYearOrDuration DateYearOrDuration::parseXsdDatetime(
 }
 
 // _____________________________________________________________________________
-DateYearOrDuration DateYearOrDuration::parseXsdDate(
+DateYearOrDuration DateYearOrDuration::parseXsdDatetime(
+    std::string_view dateString) {
+  if (auto optDate = parseXsdDatetimeImpl(dateString); optDate) {
+    return optDate.value();
+  }
+  throw DateParseException{absl::StrCat(
+      "The value ", dateString, " cannot be parsed as an `xsd:dateTime`.")};
+}
+
+// _____________________________________________________________________________
+std::optional<DateYearOrDuration>
+DateYearOrDuration::parseXsdDatetimeGetOptDate(std::string_view dateString) {
+  return parseXsdDatetimeImpl(dateString);
+}
```

Comment:
No real need for this function,
Just rename the `Impl` to the name of this function, and call it from the throwing version.